scan-command/src/test/java/io/ballerina/scan/utils/DiagnosticLogTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.scan.utils;
+
+import io.ballerina.scan.BaseTest;
+import io.ballerina.scan.RuleKind;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import static io.ballerina.scan.TestConstants.LINUX_LINE_SEPARATOR;
+import static io.ballerina.scan.TestConstants.WINDOWS_LINE_SEPARATOR;
+
+/**
+ * Diagnostic log tests.
+ *
+ * @since 0.1.0
+ */
+public class DiagnosticLogTest extends BaseTest {
+    private final Path commandOutputs = testResources.resolve("command-outputs");
+
+    @Test(description = "Test error diagnostic message")
+    public void testErrorDiagnostic() throws IOException {
```

Comment:
Check how we are implementing negative tests in here and refactor the test.

https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/mappingconstructor/MappingConstructorExprTest.java#L70-L102

Reply:
The `BCompileUtil` is used for generating diagnostic errors after compiling Single Ballerina files. However, most error messages in the scan tool are displayed before compiling the project. Only the `RuleNotFound` error is created after a compilation when external compiler plugins are engaged.

Reply:
Yeah. We can't do as in the example I provided. We can use a data provider for this if we are testing the diagnostic logger